tsl/src/chunk.c

```diff
@@ -632,6 +633,23 @@ merge_chunks_lock_upgrade_mode(void)
 	return MERGE_LOCK_ACCESS_EXCLUSIVE;
 }
 
+/*
+ * Use anonymous settings value to disable multidim merges due to a bug in the
+ * routing cache with non-aligned partitions/chunks.
+ */
+static bool
+merge_chunks_multidim_allowed(void)
+{
+	const char *multidim_allowed =
+		GetConfigOption("timescaledb.merge_chunks_multidim_allowed", true, false);
+
+	if (multidim_allowed == NULL)
+		return false;
+
+	return (pg_strcasecmp("on", multidim_allowed) == 0 ||
+			pg_strcasecmp("true", multidim_allowed) == 0);
```

Comment:
Nit: You might want to support "yes" and "1" as well, since these are usually used for boolean GUCs. Might save us a few hours in a future weird test failure.